### PR TITLE
DDP-7465 | frontend sorting

### PR DESCRIPTION
**Pull request by gmakharat**
DDP-7465

**Things have done:**

- Created payload class `Sort`.
- Deleted code which was sorting list on frontend side.
- Renamed fields to match fields coming from backend.

P.S code was written using pair programming with @Anzori-Ghurtchumelia

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/models/participant.model.ts [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+    this.reviewed = reviewed;
+    this.crSent = crSent;
+    this.crReceived = crReceived;
+    this.notes = notes;
+    this.minimalMr = minimalMr;
+    this.abstractionReady = abstractionReady;
+    this.mrNeedsAttention = mrNeedsAttention;
+    this.tissueNeedsAttention = tissueNeedsAttention;
+    this.exitDate = exitDate;
+    this.additionalValuesJson = additionalValuesJson;
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/models/participant.model.ts]: **a tip**: Lines 8-23 can be skipped.
Please see [ts-shorthand](https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd)

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts [hunk 2]
```diff
@@ -1465,179 +1470,37 @@ export class ParticipantListComponent implements OnInit {
   }
 
   public isSortField(name: string): boolean {
-    return name === this.sortField;
+    if (this.sortBy) {
+      return name === this.sortBy.innerProperty;
+    } else {
+      return false;
+    }
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts]: Lines 1473-1477 can be shortened 
like
`return this.sortBy && name === this.sortBy.innerProperty`

or even 
`return name === this.sortBy?.innerProperty`

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts [hunk 3]
```diff
@@ -1465,179 +1470,37 @@ export class ParticipantListComponent implements OnInit {
...
   }
 
   sortByColumnName(col: Filter, sortParent: string): void {
-    this.sortDir = this.sortField === col.participantColumn.name ? (this.sortDir === 'asc' ? 'desc' : 'asc') : 'asc';
-    this.sortField = col.participantColumn.name;
+    this.sortOrder = this.sortBy !== null
+          ? this.sortBy.innerProperty === col.participantColumn.name
+          ? ( this.sortOrder === 'asc' ? 'desc' : 'asc' ) : 'asc' : 'asc';
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts]: Can you please double-check whether the condition is correct ? (lines 1481-1483)

File: ddp-workspace/projects/ddp-dsm-ui/src/app/sort/sort.model.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    constructor(public type: string, public additionalType: string, public tableAlias: string,
+        public outerProperty: string, public innerProperty: string, public order: string, public activityVersions?: string[]) {
+            this.type = type;
+            this.additionalType = additionalType;
+            this.tableAlias = tableAlias;
+            this.outerProperty = outerProperty;
+            this.innerProperty = innerProperty;
+            this.order = order;
+            this.activityVersions = activityVersions;
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/sort/sort.model.ts]: see https://github.com/broadinstitute/ddp-angular/pull/1086#discussion_r817700189

File: ddp-workspace/projects/ddp-dsm-ui/src/app/sort/sort.model.spec.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { Sort } from './sort.model';
+
+describe('Sort', () => {
+  it('should create an instance', () => {
+    expect(new Sort()).toBeTruthy();
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/sort/sort.model.spec.ts]: The test fails with "arguments are not provided".

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts (hunk 3 above)
Comment (gmakharat) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts]: why did you get doubts about it?

Comment (alextanlan) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts]: It is quite hard to read (multiple conditions with ternary operators) and maintain if needed.
High possibility to make a mistake.

Comment (gmakharat) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts]: @alextanlan changed the code to bring more clarification, could you check it?

Comment (alextanlan) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts]: that is better, thank you!

**Review by alextanlan — DISMISSED** [review]

**Review by ScalaChamp — APPROVED** [review]

**Pull request merged**